### PR TITLE
refactor: Convert CardTemplatePreviewerTest.java to Kotlin

### PR DESCRIPTION
**Pull request by codingtosh**
refactor: Convert CardTemplatePreviewerTest.java to Kotlin
com.ichi2.anki.CardTemplatePreviewerTest

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt [hunk 1]
```diff
@@ -13,377 +13,304 @@
...
-        ActivityController<TestCardTemplatePreviewer> previewerController = Robolectric.buildActivity(TestCardTemplatePreviewer.class, intent).create().start().resume().visible();
-        saveControllerForCleanup((previewerController));
-        TestCardTemplatePreviewer testCardTemplatePreviewer = previewerController.get();
-        Assert.assertTrue("model change did not show up?",
-                testCardTemplatePreviewer.getDummyCard(collectionBasicModelOriginal, 0).q().contains("PREVIEWER_TEST") &&
-                        testCardTemplatePreviewer.getDummyCard(collectionBasicModelOriginal, 0).a().contains("PREVIEWER_TEST"));
+    fun testPreviewUnsavedTemplate() {
+        val modelName = "Basic"
+        val collectionBasicModelOriginal = getCurrentDatabaseModelCopy(modelName)
+        val template = collectionBasicModelOriginal.getJSONArray("tmpls").getJSONObject(0)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt:41]: could you fix the line spacing on the first line one to match the original )and any other changes) - git's having issues with the diff and that may help

File: AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt [hunk 2]
```diff
@@ -13,377 +13,304 @@
...
+        template.put("qfmt", template.getString("qfmt") + "PREVIEWER_TEST")
+        val tempModelPath = TemporaryModel.saveTempModel(targetContext, collectionBasicModelOriginal)
+        val intent = Intent(Intent.ACTION_VIEW)
+        intent.putExtra(TemporaryModel.INTENT_MODEL_FILENAME, tempModelPath)
+        intent.putExtra("index", 0)
+        var previewerController = Robolectric.buildActivity(TestCardTemplatePreviewer::class.java, intent).create().start().resume().visible()
+        saveControllerForCleanup(previewerController)
+        var testCardTemplatePreviewer = previewerController.get()
+        Assert.assertTrue(
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt:50]: similarly: could you fix the spacing to see if we can fix the diff of this method -  it'll add a small maintenance burden

File: AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt [hunk 3]
```diff
@@ -13,377 +13,304 @@
...
+        val intent = Intent(Intent.ACTION_VIEW)
+        intent.putExtra(TemporaryModel.INTENT_MODEL_FILENAME, tempModelPath)
+        val noteEditorBundle = Bundle()
+        noteEditorBundle.putBundle("editFields", getFieldsAsBundleForPreview(fields))
+        noteEditorBundle.putInt("ordinal", 0)
+        noteEditorBundle.putLong("did", 1)
+        intent.putExtra("noteEditorBundle", noteEditorBundle)
+        val testCardTemplatePreviewer = super.startActivityNormallyOpenCollectionWithIntent(TestCardTemplatePreviewer::class.java, intent)
+        MatcherAssert.assertThat("prev should not be visible", testCardTemplatePreviewer.previousButtonVisible(), CoreMatchers.`is`(false))
+        MatcherAssert.assertThat("next should not be visible", testCardTemplatePreviewer.nextButtonVisible(), CoreMatchers.`is`(false))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt:202]: import `CoreMatchers`

File: AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt [hunk 4]
```diff
@@ -13,377 +13,304 @@
...
-        intent.putExtra("noteEditorBundle", noteEditorBundle);
-
-        TestCardTemplatePreviewer testCardTemplatePreviewer = super.startActivityNormallyOpenCollectionWithIntent(TestCardTemplatePreviewer.class, intent);
-
-        assertThat("prev should not be visible", testCardTemplatePreviewer.previousButtonVisible(), is(false));
-        assertThat("next should not be visible", testCardTemplatePreviewer.nextButtonVisible(), is(false));
+    fun singleTemplateFromNoteEditorHasNoNavigation() {
+        val fields: MutableList<NoteService.NoteField?> = ArrayList()
+        fields.add(Field(0, "Hello"))
+        fields.add(Field(1, "World"))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt:190]: ideally fix spacing to how it was for readabiltiy )and with a few methods below)

File: AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt [hunk 5]
```diff
@@ -13,377 +13,304 @@
...
+        MatcherAssert.assertThat("prev should be enabled", testCardTemplatePreviewer.previousButtonEnabled(), CoreMatchers.`is`(true))
+        MatcherAssert.assertThat("next should not be enabled", testCardTemplatePreviewer.nextButtonEnabled(), CoreMatchers.`is`(false))
+        testCardTemplatePreviewer.onPreviousTemplate()
 
         // previous
-        assertThat("index is changed", testCardTemplatePreviewer.getTemplateIndex(), is(0));
-        assertThat("prev should be enabled", testCardTemplatePreviewer.previousButtonEnabled(), is(false));
-        assertThat("next should not be enabled", testCardTemplatePreviewer.nextButtonEnabled(), is(true));
+        MatcherAssert.assertThat("index is changed", testCardTemplatePreviewer.templateIndex, CoreMatchers.`is`(0))
+        MatcherAssert.assertThat("prev should be enabled", testCardTemplatePreviewer.previousButtonEnabled(), CoreMatchers.`is`(false))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt:298]: import `MatcherAssert`

File: AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt [hunk 6]
```diff
@@ -13,377 +13,304 @@
...
-            return mOrd;
-        }
-
-
-        @Nullable
-        @Override
-        public String getFieldText() {
-            return mText;
-        }
+    private inner class Field(override val ord: Int, private val text: String) : NoteService.NoteField {
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt:312]: cleanup: override fieldText in constructor and remove `text`

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/anki/CardTemplatePreviewerTest.kt:43]: nope... oh well

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
Aww it's my first real AnkiDroid contribution in 2018! https://github.com/ankidroid/Anki-Android/pull/5151#issuecomment-643510427 - took me a year and a half to write this file and the complex of PRs around it since I'd never done Android before. It lives on in a bright Kotlin future. Thanks

**Pull request merged**